nextcloudappstore/core/views.py

```diff
@@ -18,19 +20,48 @@ def get_context_data(self, **kwargs):
 class CategoryAppListView(ListView):
     model = App
     template_name = 'app/list.html'
+    allow_empty = True
 
     def get_queryset(self):
         category_id = self.kwargs['id']
         queryset = super().get_queryset()
+
         if category_id:
-            return queryset.filter(categories__id=category_id)
-        else:
-            return queryset
+            queryset = queryset.filter(categories__id=category_id)
+
+        if self.search_terms:
+            query = None
+
+            for term in self.search_terms:
```

Comment:
Ok, I think I understand now what this section means, you are doing map reduce the hard way :D

Another thing: 
* you probably want to use icontains to not depend on lower and uppercasing
* you probably want to join the search terms using & and not |, otherwise you will get more results if you add more terms when you want the opposite

Assuming that search_terms is a list of strings I'd actually write it that way (you can also get rid of the if self.search_terms):

```python
from functools import reduce
from django.db.models.functions import Lower

def create_search_query(terms):
    predicates = map(lambda t: (Q(translations__name__icontains=t) |
                                Q(translations__description__icontains=t)),
                     terms)
    return reduce(lambda x, y: x & y, predicates, Q())
```